### DISCUSSION TITLE
Issue loading assets

### DISCUSSION THREAD

**Discussion by mona09209** [Laravel-Backpack/community-forum#1085] [Q&A (Help)]
After uploaded My project to cpanel I got error in fetching basset and this is my work look like the image given so how to fix this I tried a lot of methods but nothing changed plus i edit the .env file for production and did the commands like clearing basset cache, optimize, config,...

![IMG_20240717_084110_402](https://github.com/user-attachments/assets/04115034-9488-4cd9-93ef-0f9737b927d0)

**Comment by jcastroa87** [comment]
Hello @mona09209 

If you set prod, check "APP_URL" in your .env file.

If you are using https also make sure you write it down in app url.

Then clean with
```
php artisan optimize:clear
php artisan basset:clear
```

In case this didnt work, check permissions and repeat the clean again:
```
chmod -R 755 storage
chmod -R 755 vendor
chmod -R 644 bootstrap/caches
```

And take a look in you error logs and the URLs in the network tab.

Cheers.

**Reply by mona09209** [reply]
Thanks for helping,but the issue still not solved

**Comment by pxpm** [comment]
Hey @mona09209 

please open the browser console and check the network tab for the file loading error. 

Check the url where the browser is trying to load the files. Does it match the same APP_URL in your `.env` file ? 

So for example, if you website is loading via `php artisan serve`, most likely is running on `127.0.0.1:8000` so in your in your `.env` file make sure you have `APP_URL=http://127.0.0.1:8000`

Another thing you could try is to run a `php artisan basset:check`. Please make sure your webserver is running before running that command. So if you use `php artisan serve` to launch the webserver you will need to open another console window to run that command while the server is running on the other. 

Let me know if that helps.

Cheers

**Reply by mona09209** [reply]
Sorry but the project running on host server (Cpanel) not local and this is .env content
APP_NAME=Laravel
APP_ENV=production
APP_KEY=base64:3HPn3WrB115APeNijHIwyYO+HzEFZ9DiBOjhxm6h1f0=
APP_DEBUG=false
APP_URL=https://rental.etoolabs.com


for filesystems.php

<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Default Filesystem Disk
    |--------------------------------------------------------------------------
    |
    | Here you may specify the default filesystem disk that should be used
    | by the framework. The "local" disk, as well as a variety of cloud
    | based disks are available to your application. Just store away!
    |
    */

    'default' => env('FILESYSTEM_DISK', 'local'),

    /*
    |--------------------------------------------------------------------------
    | Filesystem Disks
    |--------------------------------------------------------------------------
    |
    | Here you may configure as many filesystem "disks" as you wish, and you
    | may even configure multiple disks of the same driver. Defaults have
    | been set up for each driver as an example of the required values.
    |
    | Supported Drivers: "local", "ftp", "sftp", "s3"
    |
    */

    'disks' => [

        'local' => [
            'driver' => 'local',
            'root' => storage_path('app'),
            'throw' => false,
        ],

        'public' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'url' => env('APP_URL').'/storage',
            'visibility' => 'public',
            'throw' => false,
        ],

        'uploads' => [ // used for Backpack/CRUD (in elFinder)
            'driver' => 'local',
            'root'   => public_path('uploads'),
        ],

        'backups' => [ // used for Backpack/BackupManager
            'driver' => 'local',
            'root'   => storage_path('backups'), // that's where your backups are stored by default: storage/backups
        ],

        'storage' => [ // used for Backpack/LogManager
            'driver' => 'local',
            'root'   => storage_path(),
        ],

        's3' => [
            'driver' => 's3',
            'key' => env('AWS_ACCESS_KEY_ID'),
            'secret' => env('AWS_SECRET_ACCESS_KEY'),
            'region' => env('AWS_DEFAULT_REGION'),
            'bucket' => env('AWS_BUCKET'),
            'url' => env('AWS_URL'),
            'endpoint' => env('AWS_ENDPOINT'),
            'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
            'throw' => false,
        ],

    ],

    /*
    |--------------------------------------------------------------------------
    | Symbolic Links
    |--------------------------------------------------------------------------
    |
    | Here you may configure the symbolic links that will be created when the
    | `storage:link` Artisan command is executed. The array keys should be
    | the locations of the links and the values should be their targets.
    |
    */

    'links' => [
        public_path('storage') => storage_path('app/public'),
    ],

];

**Comment by mona09209** [comment]
What i deploy the project and edit the .env file but basset still not
fetching

**Comment by pxpm** [comment]
Cpanel .. if you don't have access to ssh it will be tricky to deploy any Laravel project, not only Backpack specific. 

If you do have access to ssh, make sure your storage is linked with `php artisan storage:link`, that `php artisan basset:check` command passes all green. 

If you don't have access to ssh, you need to run `php artisan basset:cache` on your local machine so that the assets are generated on local, and then copy those assets to your cpanel instance. 

What are you using to deploy to prod ? Fetch from github repo and run some commands like `composer install --no-dev` etc ? 

Cheers

**Comment by mona09209** [comment]
Let me explain to what i did:
1) After completed the design of front pages
2).I changed the .env settings to:

a)App_URL:https://rental.etoolabs.com
b)App_env: production
c)App_debug: false

3)Then I upload the project to cpanel host which has ( GUI ) and did
multiple commands on cpanel terminal:

php artisan optimize:clear
php artisan basset:clear
php artisan basset:cache
php artisan optimize
Php artisan storage: Link
Php artisan basset: check

Finally I got error on fetching basset even when I edit the permission for
storage directory and bootstrap.

**Reply by pxpm** [reply]
And when you run `basset:check`, what was the error ? 

After you run `php artisan basset:cache` do you have your files in `public/storage/basset` ? If you can't find `storage` in your public folder it means the symlink is not properly created. re-create it again, and if the issue persist ask your host for a solution.

Just another heads up, your `storage:link` must be created before the `basset:cache` command, otherwise basset won't be able to write to that directory. 

Alternatively, if you can't link the storage directory you can create a new basset disk pointing to your public folder directly. 

```php
// in your `config/filesystems.php` file
 'bassetDisk' => [
    'driver' => 'local',
    'root' => public_path(''),
    'url' => '/',
    'visibility' => 'public',
    'throw' => false,
],

// in your .env file: 
BASSET_DISK=bassetDisk
```
This way you shouldn't need a symlink if that's the issue in your case. 

Please, if you need my further assistance I need you to start providing more information, screenshots with the errors (paint the confidential information), what you tried etc.. at the moment I am just blind guessing, hope I am good at it 😃 

Cheers

**Comment by mona09209** [comment]
The problem solved by installing node.js in project root directory.
Thanks you all for your help 🙏!


Cheers,

**Comment by jcastroa87** [comment]
Im glad you solve the problem.

I will close the issue, but please feel free to re-open or create a new one if needed.

Cheers.
